In [147]:
import string
import pandas as pd
import numpy as np
from sklearn.model_selection import KFold

In [2]:
data = pd.read_csv('INTUITIVE1.csv')
data

In [4]:
new_columns = data.columns.values
new_columns[2] = 'text'
data.columns = new_columns

In [5]:
data

id                                             target  \
0        1  0000100001000100000000001000000000001000010001...   
1        2  0000010001000100010001001000010001000100010010...   
2        3  1000100010000100100001000100010010001000010010...   
3        4  0000010000000100100010000000100001000100010001...   
4        5  0100100000001000100001000000010010001000000001...   
...    ...                                                ...   
1113  1245  1000010000000100010010000000010001000100010001...   
1114  1246  1000000010000100100010000000010000000100010000...   
1115  1247  0100010010001000100001001000010001001000010001...   
1116  1248  1000010000001000100000001000010001001000010010...   
1117  1249  1000100001000100000001000000010010000100100010...   

                                                   text  
0     ['wmc', 'AM', 'anemia', 'sign', 'di', 'admiss'...  
1     ['VH', 'AM', 'discharg', 'summari', 'sign', 'd...  
2     ['aech', 'AM', 'pnuemonia', 'sign', 'di', 'adm...  
3     ['emh', 'AM', 'discharg', 'summari', 'sign', '...  
4     ['kfm', 'AM', 'congest', 'heart', 'failur', 'u...  
...                                                 ...  
1113  ['chh', 'AM', 'coronari', 'arteri', 'diseas', ...  
1114  ['bcvhc', 'AM', 'discharg', 'summari', 'sign',...  
1115  ['ccvgh', 'AM', 'congest', 'heart', 'failur', ...  
1116  ['mmh', 'AM', 'fever', 'unsign', 'di', 'admiss...  
1117  ['CH', 'AM', 'asthma', 'sign', 'di', 'admiss',...  

[1118 rows x 3 columns]

In [6]:
data.text.nunique()

1118

Berdasarkan dua hasil diatas (sama-sama menghasilkan 1118 baris), maka semua teks medis adalah unik. Tidak ada dua teks yang memiliki kata-kata yang persis sama.

In [134]:
table = str.maketrans('', '', string.punctuation)
tampungan = list()
for x in a:
    b = ''
    s = str(x)
    words = s.split()
    stripped = [w.translate(table) for w in words]
    for y in stripped:
        k = y.replace("‘", '').replace("’", '').replace("'", '')
        b+= k +' '
    b = b.replace("‘", '').replace("’", '').replace("'", '')
    tampungan.append(b)

In [143]:
data['teks'] = tampungan

In [145]:
data.drop('text', axis=1, inplace=True)

In [146]:
data

id                                             target  \
0        1  0000100001000100000000001000000000001000010001...   
1        2  0000010001000100010001001000010001000100010010...   
2        3  1000100010000100100001000100010010001000010010...   
3        4  0000010000000100100010000000100001000100010001...   
4        5  0100100000001000100001000000010010001000000001...   
...    ...                                                ...   
1113  1245  1000010000000100010010000000010001000100010001...   
1114  1246  1000000010000100100010000000010000000100010000...   
1115  1247  0100010010001000100001001000010001001000010001...   
1116  1248  1000010000001000100000001000010001001000010010...   
1117  1249  1000100001000100000001000000010010000100100010...   

                                                   teks  
0     wmc AM anemia sign di admiss date report statu...  
1     VH AM discharg summari sign di admiss date rep...  
2     aech AM pnuemonia sign di admiss date report s...  
3     emh AM discharg summari sign di admiss date re...  
4     kfm AM congest heart failur unsign di admiss d...  
...                                                 ...  
1113  chh AM coronari arteri diseas sign di admiss d...  
1114  bcvhc AM discharg summari sign di admiss date ...  
1115  ccvgh AM congest heart failur rule out myocard...  
1116  mmh AM fever unsign di admiss date report stat...  
1117  CH AM asthma sign di admiss date report statu ...  

[1118 rows x 3 columns]

Kolom id, target, dan teks seperti penampakan df diatas akan digunakan dalam perhitungan tf-idf.
<br> Dilanjutkan di 2_. <br>

In [148]:
data.to_csv('2_data.csv',index=False)